bt/tasks/utility/bt_call_method.cpp

```diff
@@ -61,19 +67,24 @@ BT::Status BTCallMethod::_tick(double p_delta) {
 	Object *obj = node_param->get_value(get_agent(), get_blackboard());
 	ERR_FAIL_COND_V_MSG(obj == nullptr, FAILURE, "BTCallMethod: Failed to get object: " + node_param->to_string());
 
+	const Variant delta = include_delta ? Variant(p_delta) : Variant();
 	const Variant **argptrs = nullptr;
 
-	if (args.size() > 0) {
-		argptrs = (const Variant **)alloca(sizeof(Variant *) * args.size());
-		for (int i = 0; i < args.size(); i++) {
-			argptrs[i] = &args[i];
+	int argument_count = include_delta ? args.size() + 1 : args.size();
+	if (argument_count > 0) {
+		argptrs = (const Variant **)alloca(sizeof(Variant *) * argument_count);
+		if (include_delta) {
+			argptrs[0] = &delta;
+		}
+		for (int i = 0; i < argument_count; i++) {
+			argptrs[i + 1] = &args[i];
```

Comment:
Crash due to index out of bounds when not including delta and having one or more arguments.

Reply:
While it might not crash, it writes past the allocated size and offsets the index when `include_delta == false`.
This should work: `argptrs[i + int(include_delta)] = &args[i];`

Reply:
> This should work: `argptrs[i + int(include_delta)] = &args[i];`

I tried that patch as well and for some reason it's not working -- I don't see an off-by-one error there nor with another patch I had. I'll dig further. I'm not really sure what's happening.

Reply:
Welp, it was the use of `argument_count` on the for-loop condition -- I should have used `args.size()`.